compiler/src/steps/validate-model.ts

```diff
@@ -773,6 +773,13 @@ export default async function validateModel (apiModel: model.Model, restSpec: Ma
       }
 
       context.push(`Property '${prop.name}'`)
+
+      if (prop.type.kind === 'dictionary_of' && prop.type.ordered) {
+        if (prop.name !== 'aggregations') {
```

Comment:
This check only works for the "top-level" type of properties and does not check the type-name, but I think this is sufficient for now. WDYT?

Reply:
> It seems a bit brittle, but will ensure we can easily use OrderedDictionary for all aggregations property, which is where it should always be used.

Correct!

> ... which makes me think that ideally we should have a validation that forbids Dictionary<string, Aggregation> to ensure we haven't forgotten anything.

I don't think there is a good way to detect this using a pattern. We have also cases like this:

```ts
Dictionary<string, ApiKeyAggregationContainer>
```

where we define special containers to restrict the valid aggregations to a subset.